backend/internal/auth/sessionmanager.go

```diff
@@ -66,26 +90,103 @@ func (sm *SessionManager) GetSession(sid SessionID) (Session, error) {
 	}
 
 	if time.Now().After(session.Expires) {
+		// Session expired, delete it from memory
+		sm.deleteSession(sid)
 		return Session{}, ErrSessionTimeout
 	}
 
 	return session, nil
 }
 
+func (sm *SessionManager) getSessionFromContext(ctx context.Context) Session {
+	session, ok := ctx.Value(sm.contextKey).(Session)
+	if !ok {
+		// Should be impossible
+		panic("No Session in context, software design error")
```

Comment:
you do not want to shutdown a service this way (it assumes that getting a websession from the context is the most important operation there is ignoring metrics collector, etc). Instead you can do something like:

```go
func (lh *LogoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
        // you can add FromContext method to your session manager
        sessionId, err := sm.FromContext(r.Context())
        if err != nil {
          // return error + log
        }

        sm.DeleteSession(sessionId)
        w.WriteHeader(http.StatusNoContent)	
}
```

Reply:
Fixed this, though slightly modified in that `FromContext` returns an entire `Session` object (rather than just `sessionID`. That way other endpoints can use it to make use of `FromContext` to access other `Session` fields etc.